lldb/source/Plugins/LanguageRuntime/Swift/SwiftTask.cpp

```diff
@@ -0,0 +1,62 @@
+#include "SwiftTask.h"
+#include "SwiftLanguageRuntime.h"
+#include "lldb/Target/Process.h"
+
+using namespace llvm;
+using namespace lldb;
+
+namespace lldb_private {
+
+ThreadTask::ThreadTask(tid_t tid, addr_t async_ctx, ExecutionContext &exe_ctx)
+    : Thread(exe_ctx.GetProcessRef(), tid, true),
+      m_reg_info_sp(exe_ctx.GetFrameSP()->GetRegisterContext()) {
+  m_async_ctx = async_ctx;
+  uint32_t ptr_size =
+      exe_ctx.GetTargetRef().GetArchitecture().GetAddressByteSize();
+  // A simplified description of AsyncContext. See swift/Task/ABI.h
+  // struct AsyncContext {
+  //   AsyncContext *Parent;                    // offset 0
+  //   TaskContinuationFunction *ResumeParent;  // offset 8
+  // };
+  uint32_t resume_offset = ptr_size; // offsetof(AsyncContext, ResumeParent)
+  uint32_t resume_ptr = async_ctx + resume_offset;
+  Status status;
+  m_resume_fn =
+      exe_ctx.GetProcessRef().ReadPointerFromMemory(resume_ptr, status);
```

Comment:
IIUC to apply the skip prologue trick, you would need to factor out the fragment below of `SwiftLanguageRuntime::TrySkipVirtualParentProlog` into a separate function, and say that `pc_value = m_resume_fn`:

```
  Address pc;
  Target &target = process.GetTarget();
  pc.SetLoadAddress(pc_value, &target);
  if (!pc.IsValid())
    return {};

  SymbolContext sc;
  if (!pc.CalculateSymbolContext(&sc,
                                 eSymbolContextFunction | eSymbolContextSymbol))
    return {};
  if (!sc.symbol && !sc.function)
    return {};

  auto prologue_size = sc.symbol ? sc.symbol->GetPrologueByteSize()
                                 : sc.function->GetPrologueByteSize();
  return pc_value + prologue_size;
```

Reply:
This will be added in a follow up change, which adds `language swift task select <task>`.